source/NVDAObjects/UIA/edge.py

```diff
@@ -1,26 +1,306 @@
+					childChildren=childRange.getChildren()
+				if childChildren.length==1 and UIAHandler.handler.clientObject.compareElements(child,childChildren.getElement(0)):
+					return True
+		return False
+
+	def move(self,unit,direction,endPoint=None):
+		# Skip over non-text element starts and ends
+		if not endPoint:
+			if direction>0 and unit in (textInfos.UNIT_LINE,textInfos.UNIT_PARAGRAPH):
+				return super(EdgeTextInfo,self).move(unit,direction)
+			elif direction>0:
+				res=super(EdgeTextInfo,self).move(unit,direction)
+				if res!=0:
+					# Ensure we move past the start of any elements 
+					tempInfo=self.copy()
+					while super(EdgeTextInfo,tempInfo).move(textInfos.UNIT_CHARACTER,1)!=0:
+						tempInfo.setEndPoint(self,"startToStart")
+						if tempInfo.text or tempInfo._hasEmbedded():
+							break
+						tempInfo.collapse(True)
+						self._rangeObj=tempInfo._rangeObj.clone()
+				return res
+			elif direction<0:
+				tempInfo=self.copy()
+				res=super(EdgeTextInfo,self).move(unit,direction)
+				if res!=0:
+					tempInfo.setEndPoint(self,"startToStart")
+					if not tempInfo.text and not tempInfo._hasEmbedded():
+						self.move(textInfos.UNIT_CHARACTER,-1)
+				return res
+		else:
+			tempInfo=self.copy()
+			res=tempInfo.move(unit,direction)
+			if res!=0:
+				self.setEndPoint(tempInfo,"endToEnd" if endPoint=="end" else "startToStart")
+			return res
+
+	def expand(self,unit):
+		# Ensure expanding to character/word correctly covers embedded controls
+		if unit in (textInfos.UNIT_CHARACTER,textInfos.UNIT_WORD):
+			tempInfo=self.copy()
+			tempInfo.move(textInfos.UNIT_CHARACTER,1,endPoint="end")
+			if tempInfo._hasEmbedded():
+				self.setEndPoint(tempInfo,"endToEnd")
+				return
+		super(EdgeTextInfo,self).expand(unit)
+		return
+
+	def _getControlFieldForObject(self,obj,isEmbedded=False,startOfNode=False,endOfNode=False):
+		field=super(EdgeTextInfo,self)._getControlFieldForObject(obj,isEmbedded=isEmbedded,startOfNode=startOfNode,endOfNode=endOfNode)
+		# report landmarks
+		landmark=obj.UIAElement.getCurrentPropertyValue(UIAHandler.UIA_LocalizedLandmarkTypePropertyId)
+		if landmark and (landmark!='region' or field.get('name')):
+			field['landmark']=aria.landmarkRoles.get(landmark)
+		# Edit controls must expose a name
+		if obj.role==controlTypes.ROLE_EDITABLETEXT:
+			field["name"] = obj.name
+		# For certain controls, if ARIA overrides the label, then force the field's content (value) to the label
+		# Later processing in Edge's getTextWithFields will remove descendant content from fields with a value.
+		ariaProperties=obj.UIAElement.currentAriaProperties
+		hasAriaLabel=('label=' in ariaProperties)
+		hasAriaLabelledby=('labelledby=' in ariaProperties)
+		if hasAriaLabelledby:
+			field['name']=obj.name
+		if hasAriaLabel or hasAriaLabelledby:
+			if obj.role in (controlTypes.ROLE_LINK,controlTypes.ROLE_GRAPHIC,controlTypes.ROLE_BUTTON):
+				field['value']=obj.name
+				field['alwaysReportValue']=True

```

Comment:
This `value` stuff causes links with an aria-label to read incorrectly when using quick nav, for example.